code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -218,6 +218,10 @@
 
 /obj/structure/closet/proc/can_close(mob/living/user)
 	var/turf/T = get_turf(src)
+	for (var/obj/obj in loc.contents)
+		if (istype(obj, /obj/structure/closet) && obj != src)
```

Comment:
why are you istype()ing this after typepath filtering it already. you can just do `for(var/obj/structure/closet in loc)` and do the not src test